sourcegraph/learn
# Get ESLint running again

### What should this PR do?
Resolves [DEVED-241](https://sourcegraph.atlassian.net/browse/DEVED-241) by getting ESLint up & running again. 

This may have been related to recent dependency updates in our repo, **or** in `@sourcegraph/eslint-config`, which had some dependency updates in the past ~24 hours.

Either way, we're addressing warnings about missing plugins here, as well as the incompatibility between `eslint-plugin-react` and the latest version of `eslint`. Solution for now is pinning to last major version.

### Why are we making this change?
- We definitely can't function without `eslint`!

### What are the acceptance criteria? 
- ESLint should run without error.
- Our pre-commit hook should work.

### How should this PR be tested?
- Check out the branch, remove `node_modules`, and install deps with `npm ci`
- Check that `npm run lint` runs without error.

## Pull request process

**Reviewers**:

1. Test functionality using the criteria above.
2. Offer tips for efficiency, feedback on best practices, and possible alternative approaches and things that may not have been considered.
3. For shorter, "quick" PRs, use your best judgement on #​2.
4. Use a collaborative approach and provide resources and/or context where appropriate.
5. Provide screenshots/grabs where appropriate to show findings during review.

**Reviewees**:

1. Prefer incremental and appropriately-scoped changes.
2. Leave a comment on things you want explicit feedback on.
3. Respond clearly to comments and questions.
